### PR TITLE
Add Holds migration samples for vault.

### PR DESCRIPTION
**Pull request by MattLau**

### CODE REVIEW

File: vault/vault-hold-migration-api/src/main/java/com/google/vault/chatmigration/DirectoryService.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                user =  RetryableTemplate
+                    .callWithRetry(() -> directoryService.users().get(accountId).execute());
+                users.put(accountId, user);
+            } catch (ExecutionException e) {
+                logger.log(Level.WARNING, "Unable to get email address for account Id " + accountId, e);
+            } catch (RetryException e) {
+                logger.log(Level.WARNING, "Unable to get email address for account Id " + accountId, e);
+            }
+        }
+        return user!=null?user.getPrimaryEmail():accountId;
```
Comment (grant) [inline vault/vault-hold-migration-api/src/main/java/com/google/vault/chatmigration/DirectoryService.java]: I think this line could be formatted.

File: vault/vault-hold-migration-api/src/main/java/com/google/vault/chatmigration/DuplicateHold.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      String accounts = record.get(HoldsReport.ACCOUNT_IDS);
+
+      List<HeldAccount> accountList = (accounts.equals("")) ? null :Arrays.stream(accounts.split(","))
+          .map(account -> new HeldAccount().setAccountId(account))
+          .collect(Collectors.toList());
+      boolean exceedsAccountLimit = false;
+
+      Hold hold = new Hold().setName(name + HOLD_NAME_SUFFIX).setCorpus("HANGOUTS_CHAT");
+
+      if (!"".equals(orgUnitId)) {
```
Comment (grant) [inline vault/vault-hold-migration-api/src/main/java/com/google/vault/chatmigration/DuplicateHold.java]: This is a bit backwards.
We should test if `orgUnitId` equals `""`.

Comment (MattLau) [reply vault/vault-hold-migration-api/src/main/java/com/google/vault/chatmigration/DuplicateHold.java]: Got it thanks!

**Comment by grant** [comment]
Can we also add a file `vault/vault-hold-migration-api/README.md` that describes what this sample does?

**Comment by grant** [comment]
Fixed README text and merged.

**Pull request merged**